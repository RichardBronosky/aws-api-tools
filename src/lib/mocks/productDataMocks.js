export const mockProduct1 = {
  addon_items: [
    {
      category: 'bedroom',
      delivery_type: 'K',
      list_price: 49.99,
      price: 49.99,
      quantity: 2,
      sku: '50222606',
      title: 'Twin Bunkie Boards',
      addon_required: true,
    },
  ],
  free_shipping: true,
  variation_keys: [],
  type: 'item',
  title: 'Adaire Brown Side Table',
  description:
    "Legend has it that King Arthur's sword Excalibur was forged on a mythical islanda place where grapevines were said to grow amongst wild apple trees. This meeting of wood and metal was the inspiration behind our bold new Adaire side table. Each piece is crafted with slats of acacia wood in angled patterns, creating a rustic look offset by modern metal legs and sleek handles. Customer assembly is required. Customer assembly is required.",
  delivery_type: 'O',
  catalog: 'adult',
  room_type_code: 'LR',
  single_item_room: false,
  sell_individually: true,
  mpn: '231-CS860',
  swatch_request_flag: false,
  generic_name: 'side table',
  monogram: false,
  decor: ['contemporary'],
  finish_family: ['dark wood'],
  dimensions: '23"l x 23"w x 24"h',
  finish: ['brown'],
  special_offer: true,
  style: ['square'],
  catalog_availability: { TX: true, FL: true, OOM: true, SE: true },
  category: 'tables',
  sub_category: ['End Tables'],
  pricing: {
    SE_3_list_price: 349.99,
    SE_3_sale_price: 279.99,
    default_price: 349.99,
    FL_0_list_price: 349.99,
    FL_0_sale_price: 279.99,
    FL_6_list_price: 349.99,
    FL_6_sale_price: 279.99,
    TX_5_list_price: 349.99,
    TX_5_sale_price: 279.99,
    SE_0_list_price: 349.99,
    SE_0_sale_price: 279.99,
    SE_4_list_price: 349.99,
    SE_4_sale_price: 279.99,
    TX_4_list_price: 349.99,
    TX_4_sale_price: 279.99,
    TX_6_list_price: 349.99,
    TX_6_sale_price: 279.99,
    FL_4_list_price: 349.99,
    FL_4_sale_price: 279.99,
    FL_3_list_price: 349.99,
    FL_3_sale_price: 279.99,
    TX_0_list_price: 349.99,
    TX_0_sale_price: 279.99,
    SE_2_list_price: 349.99,
    SE_2_sale_price: 279.99,
  },
  on_sale: {
    SE_3: true,
    FL_0: true,
    FL_6: true,
    TX_5: true,
    SE_0: true,
    SE_4: true,
    TX_4: true,
    TX_6: true,
    FL_4: true,
    FL_3: true,
    TX_0: true,
    SE_2: true,
  },
  strikethrough: { FL: true, OM: true, SE: true, TX: true },
  sku_promotion_description: {
    FL: 'Memorial Day Drop Ship',
    OM: 'Memorial Day Drop Ship',
    SE: 'Memorial Day Drop Ship',
    TX: 'Memorial Day Drop Ship',
  },
  ad_position: { FL: '0', OM: '0', SE: '0', TX: '0' },
  shipping_cost_code: { FL: 'VD71', SE: 'VD71', TX: 'VD71' },
  breadcrumb_url: '/furniture/living-rooms',
  you_may_also_like_items: [
    {
      sku: '22122369',
      title: 'Adaire Brown Cocktail Table',
      quantity: 1,
      label: 'you may also like',
      addon_required: false,
      pricing: {
        SE_4_list_price: 549.99,
        SE_4_sale_price: 549.99,
        default_price: 549.99,
        FL_4_list_price: 549.99,
        FL_4_sale_price: 549.99,
        SE_3_list_price: 549.99,
        SE_3_sale_price: 549.99,
        TX_6_list_price: 549.99,
        TX_6_sale_price: 549.99,
        SE_0_list_price: 549.99,
        SE_0_sale_price: 549.99,
        TX_4_list_price: 549.99,
        TX_4_sale_price: 549.99,
        SE_2_list_price: 549.99,
        SE_2_sale_price: 549.99,
        FL_3_list_price: 549.99,
        FL_3_sale_price: 549.99,
        FL_6_list_price: 549.99,
        FL_6_sale_price: 549.99,
        FL_0_list_price: 549.99,
        FL_0_sale_price: 549.99,
        TX_0_list_price: 549.99,
        TX_0_sale_price: 549.99,
        TX_5_list_price: 549.99,
        TX_5_sale_price: 549.99,
      },
      catalog_availability: { SE: true, TX: true, FL: true, OOM: true },
      delivery_type: 'O',
      category: 'tables',
      sell_individually: true,
    },
  ],
  primary_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090',
  grid_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_grid-image-item?cache-id=5414146faa5973392eccec06f9ab3e31',
  high_res_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_high-res-image-item?cache-id=9d4049916c821c28a12bfd9438164d4e',
  swatch_image:
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_swatch-image-item?cache-id=462c75c04a46ff84dc67a370bddee768',
  alternate_images: [
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-1?cache-id=d9eb8ed838154a8d6d59e699d3a8f94a',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-2?cache-id=fade3f3f0020c0462b73618a3de9ced8',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-image-3?cache-id=0dc792fe59668f81446dde858d303efa',
  ],
  alternate_grid_images: [
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-1?cache-id=72f93b04e81c83e6772bac543445fba2',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-2?cache-id=f91b555873b9780b60143b3d7a3cc826',
    'http://images.rtg-dev.com/adaire-brown-side-table_23122360_alt-grid-image-3?cache-id=068364c441833b95ca6189391cd5dec5',
  ],
  lastModified: '2019-05-23T16:13:30.917Z',
  createdAt: '2019-05-23T16:13:30.917Z',
  sku: '23122360',
}

export const mockProduct2 = {
  free_shipping: false,
  variation_keys: ['A2C6BF7083B796559132D28DC0A8228994B4A99F'],
  type: 'room',
  title: 'Laney Park Dark Gray 7 Pc Sectional',
  description:
    'The Laney Park sectional is packed with features meant to enhance your lounging experience. Each oversized seat has a smooth adjusting back mechanism for customizable seat depth, making it the perfect place for entertaining friends or relaxing with family. Contemporary and stylish, the dark gray color and flat woven textured fabric are enhanced with clean, tailored lines. 100% Polyester.',
  delivery_type: 'D',
  catalog: 'adult',
  room_type_code: 'LR',
  single_item_room: true,
  sell_individually: true,
  mpn: '114-RG270',
  swatch_request_flag: false,
  generic_name: '7 pc sectional',
  monogram: false,
  color_family: ['gray'],
  material: ['textured'],
  movement: ['stationary'],
  style: ['sectional'],
  decor: ['contemporary'],
  dimensions: '116"w x 135"d x 33"h',
  color: ['dark gray'],
  material_family: 'textured',
  piececount: 7,
  catalog_availability: { FL: true, OOM: false, SE: true, TX: true },
  category: 'livingroom',
  sub_category: ['Living Room Sets', 'Sectionals'],
  package_skus: {
    SE: [
      { sku: '12216760', quantity: '1', price: '208.43' },
      { sku: '12116768', quantity: '4', price: '172.94' },
      { sku: '12316762', quantity: '1', price: '231.63' },
      { sku: '12516766', quantity: '1', price: '68.17' },
    ],
    TX: [
      { sku: '12316762', quantity: '1', price: '231.63' },
      { sku: '12216760', quantity: '1', price: '208.43' },
      { sku: '12516766', quantity: '1', price: '68.17' },
      { sku: '12116768', quantity: '4', price: '172.94' },
    ],
    FL: [
      { sku: '12316762', quantity: '1', price: '248.65' },
      { sku: '12116768', quantity: '4', price: '185.62' },
      { sku: '12216760', quantity: '1', price: '223.71' },
      { sku: '12516766', quantity: '1', price: '73.16' },
    ],
  },
  pricing: {
    SE_0_list_price: 1299.99,
    SE_0_sale_price: 1199.99,
    default_price: 1388,
    TX_0_list_price: 1388,
    TX_0_sale_price: 1199.99,
    FL_6_list_price: 1199.99,
    FL_6_sale_price: 1199.99,
    TX_6_list_price: 1388,
    TX_6_sale_price: 1199.99,
    TX_5_list_price: 1388,
    TX_5_sale_price: 1199.99,
    SE_3_list_price: 1199.99,
    SE_3_sale_price: 1199.99,
    FL_4_list_price: 1199.99,
    FL_4_sale_price: 1199.99,
    FL_3_list_price: 1299.99,
    FL_3_sale_price: 1288,
    SE_2_list_price: 1299.99,
    SE_2_sale_price: 1199.99,
    TX_4_list_price: 1388,
    TX_4_sale_price: 1199.99,
    SE_4_list_price: 1199.99,
    SE_4_sale_price: 1199.99,
    FL_0_list_price: 1299.99,
    FL_0_sale_price: 1288,
  },
  on_sale: {
    SE_0: true,
    TX_0: true,
    FL_6: false,
    TX_6: true,
    TX_5: true,
    SE_3: false,
    FL_4: false,
    FL_3: true,
    SE_2: true,
    TX_4: true,
    SE_4: false,
    FL_0: true,
  },
  shipping_cost_code: { FL: 'ROOM', SE: 'ROOM', TX: 'ROOM' },
  is_featured: { FL: true, SE: true, TX: true },
  variations: {
    color: [
      {
        sku: '1141677P',
        title: 'Laney Park Dark Gray 7 Pc Sectional',
        catalog_availability: { FL: true, OOM: false, SE: true, TX: true },
        variation_value: 'Dark Gray',
        image:
          'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_swatch-image-room?cache-id=b574d3b4897f3e9848ffdec84989abe3',
      },
      {
        sku: '1141676P',
        title: 'Laney Park Light Gray 7 Pc Sectional',
        catalog_availability: { TX: true, FL: true, OOM: false, SE: true },
        variation_value: 'Light Gray',
        image:
          'http://images.rtg-dev.com/laney-park-light-gray-7-pc-sectional_1141676P_swatch-image-room?cache-id=4623cf4bc97371730ba3c414794da78c',
      },
    ],
  },
  breadcrumb_url: '/furniture/living-rooms',
  primary_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
  grid_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_grid-image-room?cache-id=ae141488ad62b690ff8edb6ba7635c0a',
  high_res_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_high-res-image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
  swatch_image:
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_swatch-image-room?cache-id=b574d3b4897f3e9848ffdec84989abe3',
  alternate_images: [
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-1?cache-id=bec562965fec72af770c2db65c72e1ba',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-2?cache-id=6035fedf0cf68d55a8b118999fbb2bb5',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-3?cache-id=322a89d2f5eb4b25208d1beebc4c5d25',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-image-4?cache-id=91df28c6afd4eb7720750245bfa2d131',
  ],
  alternate_grid_images: [
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-1?cache-id=a638c2b14ccff7a4e9af1100ea21ed52',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-2?cache-id=24dc7ae0b65f2fc9c3d7a18a122aca22',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-3?cache-id=cb24d99a79ab1614820e9b8b044bdc48',
    'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_alt-grid-image-4?cache-id=898ccb8827b2499b3893f69959215f22',
  ],
  lastModified: '2019-05-23T16:16:03.416Z',
  createdAt: '2019-05-23T16:16:03.416Z',
  sku: '1141677P',
}

export const mockAddonItem = {
  category: 'bedroom',
  delivery_type: 'K',
  list_price: 49.99,
  price: 49.99,
  quantity: 2,
  sku: '50222606',
  title: 'Twin Bunkie Boards',
}

export const mockActiveAddon = {
  age: ['child'],
  breadcrumb_url: '/furniture/kids-and-teens',
  catalog: 'kids',
  catalog_availability: [{ FL: true }, { OOM: true }, { SE: true }, { TX: true }],
  category: 'bedroom',
  createdAt: '2019-07-03T16:44:10.171Z',
  delivery_type: 'K',
  description: 'Twin Bunkie Boards Twin Bunkie Boards',
  free_shipping: false,
  generic_name: 'twin bunkie boards',
  grid_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_grid-image-item?cache-id=1a3495eb7cd46f299bce08a82dffd38f',
  high_res_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_high-res-image-item?cache-id=ce8aac5dc0808e4139d59d4ef5d2b95e',
  lastModified: '2019-07-03T16:44:10.171Z',
  monogram: false,
  mpn: '502-CS106',
  on_sale: [
    { FL_0: false },
    { FL_3: false },
    { FL_4: false },
    { FL_6: false },
    { SE_0: false },
    { SE_2: false },
    { SE_3: false },
    { SE_4: false },
    { TX_0: false },
    { TX_4: false },
    { TX_5: false },
    { TX_6: false },
  ],
  pricing: [
    { FL_0_list_price: 49.99 },
    { FL_0_sale_price: 49.99 },
    { FL_3_list_price: 49.99 },
    { FL_3_sale_price: 49.99 },
    { FL_4_list_price: 49.99 },
    { FL_4_sale_price: 49.99 },
    { FL_6_list_price: 49.99 },
    { FL_6_sale_price: 49.99 },
    { SE_0_list_price: 49.99 },
    { SE_0_sale_price: 49.99 },
    { SE_2_list_price: 49.99 },
    { SE_2_sale_price: 49.99 },
    { SE_3_list_price: 49.99 },
    { SE_3_sale_price: 49.99 },
    { SE_4_list_price: 49.99 },
    { SE_4_sale_price: 49.99 },
    { TX_0_list_price: 49.99 },
    { TX_0_sale_price: 49.99 },
    { TX_4_list_price: 49.99 },
    { TX_4_sale_price: 49.99 },
    { TX_5_list_price: 49.99 },
    { TX_5_sale_price: 49.99 },
    { TX_6_list_price: 49.99 },
    { TX_6_sale_price: 49.99 },
    { default_price: 49.99 },
  ],
  primary_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_image-item?cache-id=fbd26fd0a99f3b1e470cd31d00143223',
  quantity: 2,
  sell_individually: true,
  shipping_cost_code: [{ FL: 'ITEM' }, { SE: 'ITEM' }, { TX: 'ITEM' }],
  single_item_room: false,
  size: 'twin',
  sku: '50222606',
  sub_category: ['Bunkie Boards'],
  swatch_image:
    'https://images.rtg-dev.com/twin-bunkie-boards_50222606_swatch-image-item?cache-id=f69172c21fe719d82fbb41982a2213f7',
  swatch_request_flag: false,
  title: 'Twin Bunkie Boards',
  type: 'item',
  variation_keys: [],
}
