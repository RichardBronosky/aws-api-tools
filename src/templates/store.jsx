import React from 'react'
import Helmet from 'react-helmet'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import Layout from '../components/layout'
import Breadcrumb from '../components/shared/breadcrumbs'
import RTGLink from '../components/shared/link'
import { decodeHtml, slugify, weekdays, months, getStandardTime } from '../lib/helpers/string-helper'
import { contentfulImage } from '../lib/helpers/contentful'
import { getStoreHours, getStoreName, getStoreInfoFromFeed } from '../lib/helpers/store-locator'
import { expandState } from '../lib/helpers/geo-location'
import { setupAnalytics } from '../lib/helpers/google-tag-manager'
import StoreInfo from '../components/store-locator/store-info'
import '../assets/css/pages/store-location.sass'

class Store extends React.Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedMarker: {},
  }

  componentDidMount() {
    const store = this.props.pageContext.store
    setupAnalytics({ pageData: { type: 'store', title: getStoreName(store), path: window.location.pathname } })
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generateStoreCrumbs = store => {
    let crumbs = []
    crumbs.push(this.createCrumb('Find a Showroom', '/stores'))
    if (store.state) {
      crumbs.push(
        this.createCrumb(
          `Find a Showroom - ${ expandState(store.state) }`,
          `/stores/${ slugify(expandState(store.state)) }`
        )
      )
    }
    crumbs.push(
      this.createCrumb(
        decodeHtml(`${ store.city }, ${ store.state }${ store.storeName ? ` (${ store.storeName })` : '' }`),
        window.location.pathname
      )
    )
    return {
      crumbs: crumbs,
    }
  }

  onLocationClick = (props, marker) => {
    const store = this.props.pageContext.store
    const storeInfoFromFeed = getStoreInfoFromFeed(store.storeNumber)
    let fullMarker = {
      ...props,
      title: store.address1,
      name: getStoreName(store),
      lat: store.location.lat,
      lng: store.location.lon,
      city: store.city,
      state: store.state,
      storeNumber: store.storeNumber,
      zip: store.zip,
      phone: storeInfoFromFeed.phone.replace('/', '-'),
      hours: getStoreHours(storeInfoFromFeed.store_hours),
    }
    this.setState({
      selectedMarker: fullMarker,
      activeMarker: marker,
      showingInfoWindow: true,
    })
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  getTodayOpen = store => {
    const currentDay = weekdays[new Date().getDay()]
    const month = months[new Date().getMonth()].substr(0, 3)
    const day = new Date().getDate()
    const storeOpen = store.store_hours[`${ currentDay.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ currentDay.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ currentDay }, ${ month } ${ day }: ${
        storeOpen ? `Open from ${ getStandardTime(storeOpen) } to ${ getStandardTime(storeClose) }` : 'Closed'
      }`
    } else {
      ;`${ currentDay }, ${ month } ${ day }`
    }
  }

  getHours = (store, day) => {
    const storeOpen = store.store_hours[`${ day.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ day.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ getStandardTime(storeOpen) } - ${ getStandardTime(storeClose) }`
    }
  }

  render() {
    const { activeMarker, selectedMarker, showingInfoWindow } = this.state
    const store = this.props.pageContext.store
    const address = `${ store.address1 }${ store.address2 ? `, ${ store.address2 }` : '' }, ${ store.city }, ${ store.state } ${
      store.zip
    } `
    const storeInfoFromFeed = getStoreInfoFromFeed(store && store.storeNumber)
    if (store) {
      let storeTypeTitle = `${ store.city }, ${ store.state }`
      if (store.storeType === 'Showroom' || store.storeType === 'Express') {
        storeTypeTitle = storeTypeTitle + ' Furniture & Mattress Store'
      } else if (store.storeType === 'Kids') {
        storeTypeTitle = storeTypeTitle + ' Kids Furniture & Mattress Store'
      } else if (store.storeType === 'Outlet') {
        storeTypeTitle = storeTypeTitle + ' Affordable Furniture Outlet Store'
      } else if (store.storeType === 'Outdoor') {
        storeTypeTitle = storeTypeTitle + ' Patio Furniture: Wicker, Teak Outdoor Furniture Sets'
      }
      return (
        <Layout { ...this.props }>
          <Helmet title={ storeTypeTitle } />
          { store.seo && (
            <>
              { store.seo.pageTitle && <Helmet title={ store.seo.pageTitle } /> }
              { store.seo.canonical && (
                <Helmet
                  link={ [
                    {
                      rel: 'canonical',
                      href: store.seo.canonical,
                    },
                  ] }
                />
              ) }
              { store.seo.metaDescription && (
                <Helmet
                  meta={ [
                    {
                      name: 'description',
                      content: store.seo.metaDescription.metaDescription,
                    },
                  ] }
                />
              ) }
              { store.seo.metaRobots && (
                <Helmet
                  meta={ [
                    {
                      name: 'robots',
                      content: store.seo.metaRobots,
                    },
                  ] }
                />
              ) }
              { store.seo.jsonLdSchema && (
                <Helmet
                  script={ [
                    {
                      type: 'application/ld+json',
                      innerHTML: store.seo.jsonLdSchema.jsonLdSchema,
                    },
                  ] }
                />
              ) }
            </>
          ) }
          <section className="seo-page-heading full-width hide show">
            <div className="grid-container ">
              <div className="grid-x grid-margin-x">
                <div className="cell small-12">
                  <h1>
                    { store.seo && store.seo.pageHeading
                      ? store.seo.pageHeading
                      : `${ storeTypeTitle }${ store.storeName ? ` (${ store.storeName })` : '' }` }
                  </h1>
                </div>
              </div>
            </div>
          </section>
          <div className="stores-breadcrumb">
            <Breadcrumb data={ this.generateStoreCrumbs(store) } includeHeading={ true } />
          </div>
          <div className="store-location">
            <div className="loc-info">
              { address }
              <RTGLink
                data={ {
                  url: `tel:${ storeInfoFromFeed.phone.replace('/', '-') }`,
                  title: 'Call store',
                  category: 'store page',
                  action: 'Call click',
                  label: 'Call',
                } }
                className="phone-link"
              >
                { storeInfoFromFeed.phone.replace('/', '-') }
              </RTGLink>
              <span className="today">{ this.getTodayOpen(storeInfoFromFeed) }</span>
            </div>
            <div className="grid-x map-container">
              <Map
                initialCenter={ { lat: store.location.lat, lng: store.location.lon } }
                center={ { lat: store.location.lat, lng: store.location.lon } }
                className="map"
                google={ this.props.google }
                zoom={ 11 }
                onClick={ this.onMapClicked }
              >
                <Marker
                  position={ { lat: store.location.lat, lng: store.location.lon } }
                  title={ `${ store.city }${ store.storeName ? ` - ${ store.storeName }` : '' } ${ store.storeType }` }
                  icon={ {
                    url: `http://maps.google.com/mapfiles/markerA.png`,
                  } }
                  onClick={ this.onLocationClick }
                />
                <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
                  <StoreInfo selectedMarker={ selectedMarker } />
                </InfoWindow>
              </Map>
              { store.storeImage && (
                <div className="store-image-container">
                  <img className="store-image" alt={ storeTypeTitle } src={ contentfulImage(store.storeImage.file.url) } />
                </div>
              ) }
              { storeInfoFromFeed.store_hours && (
                <div className="section-container">
                  <h2 className="section-heading">Store Hours:</h2>
                  { weekdays.map(day => (
                    <div className="hour" key={ day }>
                      { storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">{ this.getHours(storeInfoFromFeed, day) }</span>
                        </>
                      ) }
                      { !storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">Closed</span>
                        </>
                      ) }
                    </div>
                  )) }
                  { store.specialClosings && (
                    <span
                      className="special-closings"
                      dangerouslySetInnerHTML={ { __html: store.specialClosings.markdown.childMarkdownRemark.html } }
                    />
                  ) }
                </div>
              ) }
              { store.directions && (
                <div className="section-container">
                  <h2 className="section-heading">Store Location / Address:</h2>
                  <span>{ address }</span>
                  <br />
                  <span
                    className="directions"
                    dangerouslySetInnerHTML={ { __html: store.directions.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.neighborhoodsNearby && store.neighborhoodsNearby.length > 0 && (
                <div className="section-container">
                  <h2 className="section-heading">Neighborhoods Nearby:</h2>
                  <ul>
                    { store.neighborhoodsNearby.map((hood, index) => (
                      <li key={ index } className="hood">
                        { hood }
                      </li>
                    )) }
                  </ul>
                </div>
              ) }
              { store.description && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">Store Description:</h2>
                  <span
                    className="description"
                    dangerouslySetInnerHTML={ { __html: store.description.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.faqs && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">FAQs:</h2>
                  <span
                    className="faqs"
                    dangerouslySetInnerHTML={ { __html: store.faqs.markdown.childMarkdownRemark.html } }
                  />
                </div>
              ) }
            </div>
          </div>
        </Layout>
      )
    } else {
      return null
    }
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(Store)
