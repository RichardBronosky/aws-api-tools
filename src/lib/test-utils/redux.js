import configureMockStore from 'redux-mock-store'
import * as reduxStore from '../../redux/store'

const mockStore = configureMockStore([])

export const mockStoreDispatch = initialState => {
  reduxStore.store = mockStore(initialState)
  reduxStore.store.dispatch = jest.fn()
}
