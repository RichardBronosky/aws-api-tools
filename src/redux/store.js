import React from 'react'
import { Provider } from 'react-redux'
import { createStore, compose, combineReducers } from 'redux'

import global from './modules/global'
import location from './modules/location'
import cart from './modules/cart'
import checkout from './modules/checkout'

const rootReducer = combineReducers({
  global,
  location,
  cart,
  checkout,
})

// Redux DevTools
const windowObject = typeof window !== 'undefined' && window
const reduxDevTools =
  process.env.NODE_ENV === 'development' && windowObject.devToolsExtension
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : _ => _

const configureStore = initialState => createStore(rootReducer, initialState, compose(reduxDevTools))

export const store = configureStore()

// eslint-disable-next-line react/display-name,react/prop-types
export default ({ element }) => <Provider store={ store }>{ element }</Provider>
