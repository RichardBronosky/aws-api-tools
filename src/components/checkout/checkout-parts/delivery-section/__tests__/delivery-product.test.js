import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../../../redux/store'
import DeliveryProduct from '../delivery-product'
import { mockProduct1 } from '../../../../../lib/mocks/productDataMocks'

describe('DeliveryProduct', () => {
  const testProps = {
    product: mockProduct1,
    productCount: 2,
    quantity: 2,
    index: 1,
    noImage: false,
    requiredAddon: false,
  }

  it('renders correctly when product is not last index of cart', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryProduct { ...testProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when product is last index of cart', () => {
    const newTestProps = {
      ...testProps,
      index: 2,
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryProduct { ...newTestProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when product does not have an image', () => {
    const newTestProps = {
      ...testProps,
      noImage: true,
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryProduct { ...newTestProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when product is a required add-on', () => {
    const newTestProps = {
      ...testProps,
      requiredAddon: true,
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryProduct { ...newTestProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
