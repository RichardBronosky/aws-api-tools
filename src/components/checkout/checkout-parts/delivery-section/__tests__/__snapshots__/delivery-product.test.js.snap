// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryProduct renders correctly when product does not have an image 1`] = `
<div
  className="delivery-product"
>
  <div
    className="product-container grid-x"
  >
    <div
      className="product-info small-9"
    >
      <a
        aria-label="Adaire Brown Side Table"
        disabled={false}
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href="/furniture/product/adaire-brown-side-table/23122360"
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <p>
          Adaire Brown Side Table
        </p>
      </a>
    </div>
    <div
      className="quantity small-2"
    >
      QTY: 
       2
    </div>
  </div>
</div>
`;

exports[`DeliveryProduct renders correctly when product is a required add-on 1`] = `
<div
  className="delivery-product"
>
  <div
    className="product-container grid-x"
  >
    <div
      className="product-image-container small-2 large-4"
    />
    <div
      className="product-info small-6 large-5"
    >
      <a
        aria-label="Adaire Brown Side Table"
        disabled={true}
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href={null}
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <p>
          Adaire Brown Side Table
        </p>
      </a>
    </div>
    <div
      className="quantity small-2"
    >
      QTY: 
       2
    </div>
  </div>
</div>
`;

exports[`DeliveryProduct renders correctly when product is last index of cart 1`] = `
<div
  className="delivery-product separator"
>
  <div
    className="product-container grid-x"
  >
    <div
      className="product-image-container small-2 large-4"
    >
      <a
        aria-label="Adaire Brown Side Table"
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href="/furniture/product/adaire-brown-side-table/23122360"
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <img
          alt="Adaire Brown Side Table"
          className="product-image"
          src="http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090&h=150"
        />
      </a>
    </div>
    <div
      className="product-info small-6 large-5"
    >
      <a
        aria-label="Adaire Brown Side Table"
        disabled={false}
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href="/furniture/product/adaire-brown-side-table/23122360"
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <p>
          Adaire Brown Side Table
        </p>
      </a>
    </div>
    <div
      className="quantity small-2"
    >
      QTY: 
       2
    </div>
  </div>
</div>
`;

exports[`DeliveryProduct renders correctly when product is not last index of cart 1`] = `
<div
  className="delivery-product"
>
  <div
    className="product-container grid-x"
  >
    <div
      className="product-image-container small-2 large-4"
    >
      <a
        aria-label="Adaire Brown Side Table"
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href="/furniture/product/adaire-brown-side-table/23122360"
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <img
          alt="Adaire Brown Side Table"
          className="product-image"
          src="http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090&h=150"
        />
      </a>
    </div>
    <div
      className="product-info small-6 large-5"
    >
      <a
        aria-label="Adaire Brown Side Table"
        disabled={false}
        gtm-action={null}
        gtm-category={null}
        gtm-label="/furniture/product/adaire-brown-side-table/23122360"
        gtm-value={null}
        href="/furniture/product/adaire-brown-side-table/23122360"
        onClick={[Function]}
        title="Adaire Brown Side Table"
      >
        <p>
          Adaire Brown Side Table
        </p>
      </a>
    </div>
    <div
      className="quantity small-2"
    >
      QTY: 
       2
    </div>
  </div>
</div>
`;
